components/eventing-controller/pkg/handlers/nats_test.go

```diff
@@ -1072,96 +660,72 @@ func TestRetryUsingCESDK(t *testing.T) {
 	g := NewWithT(t)
 	natsServer, natsPort := startNATSServer()
 	defer natsServer.Shutdown()
-	defaultLogger := getLogger(t, kymalogger.INFO)
+	defaultLogger := getLogger(g, kymalogger.INFO)
 	// The reconnect configs should be large enough to cover the NATS server restart period
 	natsConfig := env.NatsConfig{
 		URL:           natsServer.ClientURL(),
 		MaxReconnects: 10,
 		ReconnectWait: 3 * time.Second,
 	}
-	natsClient := NewNats(natsConfig, env.DefaultSubscriptionConfig{MaxInFlightMessages: 10}, nil, defaultLogger)
-	g.Expect(natsClient.Initialize(env.Config{})).To(BeNil())
+	natsBackend := NewNats(natsConfig, env.DefaultSubscriptionConfig{MaxInFlightMessages: 10}, nil, defaultLogger)
+	g.Expect(natsBackend.Initialize(env.Config{})).Should(Succeed())
 
-	subscriber, subscriberPort := startSubscriber()
+	subscriber, _ := startSubscriber()
 	defer subscriber.Shutdown()
-	subscriberReceiveURL := fmt.Sprintf("http://127.0.0.1:%d/store", subscriberPort)
-	subscriberCheckURL := fmt.Sprintf("http://127.0.0.1:%d/check", subscriberPort)
-	// Check subscriber is running or not by checking the store
-	g.Expect(subscriber.CheckEvent("", subscriberCheckURL)).To(BeNil())
+	g.Expect(subscriber.IsRunning()).To(BeTrue())
 
 	// Create a subscription
 	sub := eventingtesting.NewSubscription("sub", "foo", eventingtesting.WithNotCleanEventTypeFilter)
-	sub.Spec.Sink = subscriberReceiveURL
+	sub.Spec.Sink = subscriber.GetSinkURL()
 	cleaner := createEventTypeCleaner(eventingtesting.EventTypePrefix, eventingtesting.ApplicationName, defaultLogger)
-	_, err := natsClient.SyncSubscription(sub, cleaner)
+	_, err := natsBackend.SyncSubscription(sub, cleaner)
 	g.Expect(err).To(BeNil())
 
 	ev1data := "sampledata"
-	g.Expect(SendEventToNATS(natsClient, ev1data)).To(Succeed())
+	g.Expect(SendEventToNATS(natsBackend, ev1data)).Should(Succeed())
 	expectedEv1Data := fmt.Sprintf("\"%s\"", ev1data)
-	g.Expect(subscriber.CheckEvent(expectedEv1Data, subscriberCheckURL)).To(Succeed())
+	g.Expect(subscriber.CheckEvent(expectedEv1Data)).Should(Succeed())
 
 	natsServer.Shutdown()
 	g.Eventually(func() bool {
-		return natsClient.connection.IsConnected()
+		return natsBackend.connection.IsConnected()
 	}, 60*time.Second, 2*time.Second).Should(BeFalse())
 	_ = eventingtesting.RunNatsServerOnPort(natsPort)
 	g.Eventually(func() bool {
-		return natsClient.connection.IsConnected()
+		return natsBackend.connection.IsConnected()
 	}, 60*time.Second, 2*time.Second).Should(BeTrue())
 
 	// After reconnect, event delivery should work again
 	ev2data := "newsampledata"
-	g.Expect(SendEventToNATS(natsClient, ev2data)).To(Succeed())
+	g.Expect(SendEventToNATS(natsBackend, ev2data)).Should(Succeed())
 	expectedEv2Data := fmt.Sprintf("\"%s\"", ev2data)
-	g.Expect(subscriber.CheckEvent(expectedEv2Data, subscriberCheckURL)).To(Succeed())
+	g.Expect(subscriber.CheckEvent(expectedEv2Data)).Should(Succeed())
 }
 
 func startNATSServer() (*server.Server, int) {
```

Comment:
Do we still need to return the int NATS port?

Reply:
It's used in one test, and we can't get it back from the server.